nvme.c

```diff
@@ -3075,11 +3084,43 @@ static int create_ns(int argc, char **argv, struct command *cmd, struct plugin *
 		return -EINVAL;
 	}
 
-	err = parse_lba_num_si(dev, "nsze", cfg.nsze_si, cfg.flbas, &cfg.nsze);
+	id = nvme_alloc(sizeof(*id));
```

Comment:
The allocated `id` memory is not freed after the usage. By the way is it necessary to allocate really since seems just use as the `id` variable but not a pointer?

Reply:
The pointer is not freed but the nvme process terminates right after the command execution so any memory leak is short-lived, but I can add a free if people prefer, not a problem for me. About the calls to malloc(), those two structures (id and gr_list) are quite big, in the order of a few kilobytes, so it comes natural to me to allocate them on the heap. But, again, it's probably not really necessary and I have not a strong reason to defend it.

Reply:
Just do this here:

```c
_cleanup_free_ struct nvme_id_ctrl *id = NULL;
_cleanup_free_ struct nvme_id_ns_granularity_list *gr_list = NULL;
```

and we should be fine. The reason why we really take care of freeing memory is to be able to use valgrind to figure out if we are leaking memory somewhere where it matters. Keeping the house clean is the motto :)

Reply:
And yes these structures should be allocated via `nvme_alloc` because of the alignment constraints of certain devices.

Reply:
I see. Thanks for your responses.